### PR TITLE
ref: update exp init --type value

### LINKED ISSUE #3656: ref: `exp init --type=dl` no longer exists
... we missed a cmd ref update somewhere because it's still mentioned in a few places of https://dvc.org/doc/command-reference/exp/init

_Originally posted by @jorgeorpinel in https://github.com/iterative/dvc.org/issues/3430#issuecomment-1154944423_

- [ ] Update to `checkpoints` anywhere the option/value are mentioned.

### PR DESCRIPTION
**Pull request by jorgeorpinel**
Closes #3656

### CODE REVIEW

**Comment by dberenbaum** [comment]
Looks like I missed this following https://github.com/iterative/dvc/pull/7703 🤦 . Added some additional changes from that PR. @jorgeorpinel PTAL 🙏 .

**Review by dberenbaum — APPROVED** [review]

File: content/docs/command-reference/exp/init.md [hunk 1]
```diff
@@ -102,10 +102,9 @@ $ dvc exp init './another_script.sh $MYENVVAR'
 - `--type` - selects the type of the stage to create. Currently it provides two
   alternatives: `checkpoint` and `default` (no need to specify this one).
```
Comment (jorgeorpinel) [inline content/docs/command-reference/exp/init.md:104]:
```suggestion
  alternatives: `checkpoint` (supports logging
  [checkpoints](/doc/command-reference/exp/run#checkpoints) during model
  training) and `default` (no need to specify this).
```

File: content/docs/command-reference/exp/init.md [hunk 2]
```diff
@@ -102,10 +102,9 @@ $ dvc exp init './another_script.sh $MYENVVAR'
...
 
-  `dl` stages are intended for use in deep-learning scenarios, where metrics and
-  plots are tracked with [DVCLive](/doc/dvclive). This also supports logging
+  `checkpoint` supports logging
   [checkpoints](/doc/command-reference/exp/run#checkpoints) during the training
```
Comment (jorgeorpinel) [inline content/docs/command-reference/exp/init.md:106]:
```suggestion
```

File: content/docs/command-reference/exp/init.md [hunk 3]
```diff
@@ -102,10 +102,9 @@ $ dvc exp init './another_script.sh $MYENVVAR'
...
-  of DL models.
+  of models.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/exp/init.md:107]:
```suggestion
```

**Review by jorgeorpinel — COMMENTED** [review]
`--type` updates

File: content/docs/command-reference/exp/init.md [hunk 4]
```diff
@@ -100,12 +100,8 @@ $ dvc exp init './another_script.sh $MYENVVAR'
   `dvc exp run`).
 
 - `--type` - selects the type of the stage to create. Currently it provides two
-  alternatives: `dl` and `default` (no need to specify this one).
+  alternatives: `checkpoint` and `default` (no need to specify this one).
```
Comment (jorgeorpinel) [inline content/docs/command-reference/exp/init.md:104]:
```suggestion
  alternatives: `checkpoint` (supports logging
  [checkpoints](/doc/command-reference/exp/run#checkpoints) during model
  training) and `default` (no need to specify this).
```

File: content/docs/command-reference/exp/init.md [hunk 5]
```diff
@@ -135,7 +132,8 @@ $ dvc exp init './another_script.sh $MYENVVAR'
 
 - `--live` - set the path to the directory where the metrics and plots
   [produced by DVCLive](https://dvc.org/doc/dvclive/dvclive-with-dvc#outputs)
-  will be found.
+  will be found. If `--live` is provided, it will override the default values
+  for `--metrics` and `--plots`.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/exp/init.md:136]:
```suggestion
  will be found. Overrides the default values for `--metrics` and `--plots`.
```

**Pull request merged**